### ISSUE TITLE
Character is not on given coordinates

### ISSUE THREAD

**Issue by Mteuahasan** [Annoraaq/grid-engine#224]
I am facing an issue making a multiplayer game with Grid Engine. Here is what I'm doing :

1 - Moving my local player with the `move()` function.
2 - Getting the new position with the `getPosition` method.
3 - Sending the new position to other clients
4 - When receiving new position : calling `moveTo` to the proper character.

Doing so, I can observe an offset between received position and the actual position of the sprite on the map.

For example, if I receive these coords : x: 10 and y: 10, I can see that the player is moving and the coordinates that I can get with `getPosition` are the proper ones. However, the sprite of the character is not located in x: 10 / y: 10.

FIY, I tried to send the coordinates at different times always with the same result : right after calling `move(Direction)` and in the Observable `positionChangeFinished`.

Also, is there a way to subscribe to this observable only for one character ? (as I don't need it for every characters, maybe subscribing directly on a "character" object).

I saw that in the Troubleshooting section of the docs, this could happen if Phaser physics are enabled but I did not specify any physics config in my base Phaser config object. Is there a default physics engine on Phaser ? (could not find an answer to this question).

Last, is there any way to disable collisions between characters but not with the furnitures ?

Thanks !

**Comment by Mteuahasan** [comment]
Indeed, the position if correct but not the sprite. In fact, I think that you can forget everything I wrote about sending and receiving positions as only my local user seems to have this problem.

I did not try to play around with offsets as indeed, the position is initially correct then change after a movement is made.

I guess that I have no physics as I use [Phaser.GameObjects.Sprite](https://photonstorm.github.io/phaser3-docs/Phaser.GameObjects.Sprite.html).

Here is a video of the issue (at the moment of the video the offset has already happened). You can see that the coordinate that I got with getPosition has an offset with the position that I log when I click on the user's tile.

FYI, I log : `Math.floor(pointer.worldX / BaseTileWidth)` (best way I found to get the proper coordinates from 'pointerup' event. Works great with my `teleport` method using Grid Engine's `setPosition`

https://user-images.githubusercontent.com/5517306/153560591-054e9b1d-406d-479c-8c82-13786731de0d.mov

Thanks a lot for you responses !

**Comment by Annoraaq** [comment]
Since I can't see how it is supposed to look like: is the sprite misplaced by a whole tile or only by a fraction of it?

Is it possible for you to provide a minimum example that has this behavior? I can't reproduce this yet.

**Comment by Mteuahasan** [comment]
It looks like it is misplaced by a full tile.

I did a quick and dirty fix that re-sync the sprite like this :

```
const spriteX = Math.round(user.container.x / BaseTileWidth + user.container.width / 2)
const spriteY = Math.round(user.container.y / BaseTileWidth + user.container.height / 2)
if (user.position.x !== spriteX || user.position.y !== spriteY) {
    user.container.setX(user.position.x * BaseTileWidth)
    user.container.setY(user.position.y * BaseTileWidth)
}
```

I'll try to make an example but as my application is quite huge it may take some time to make it mini.
I'll keep you posted, thanks !

**Comment by Annoraaq** [comment]
> It looks like it is misplaced by a full tile.

So that makes it unlikely (while not impossible) that it has to do with Phasers physics engine.

It can either be a bug in GridEngine or some part of your game is changing the coordinates of the sprite directly at some point. Hard to say.

**Comment by Annoraaq** [comment]
Closing this due to inactivity.
